pyat/at/physics/dynamic_aperture.py

```diff
@@ -0,0 +1,1012 @@
+        ranges are limited to [-1 1] in any case without notice.
+        :return:
+        """
+        # disable parallel computation
+        init_parcomp = self.parallel_computation
+        self.parallel_computation = False
+
+        # relevant_planes = []
+        # if self.mode:
+        #     relevant_planes = self.mode.split('-', 2)
+        coord0 = {'x': 1e-6, 'xp': 0.0, 'y': 1e-6, 'yp': 0.0, 'delta': 0.0, 'ct': 0.0}
+
+        for p in self.planes:
+            if self.n_points[p] > 1:
+
+                step = (self.dict_def_range[p][1] - self.dict_def_range[p][0]) / self.search_divider
+
+                coord = {'x': 1e-6, 'xp': 0.0, 'y': 1e-6, 'yp': 0.0, 'delta': 0.0, 'ct': 0.0} # copy.deepcopy(coord0)
+                while self.test_survived(coord)[0] and coord[p]< 1.0:
+                    coord[p] += step
+                self.dict_def_range[p][1] = coord[p]+step
+
+                coord = {'x': 1e-6, 'xp': 0.0, 'y': 1e-6, 'yp': 0.0, 'delta': 0.0, 'ct': 0.0} # copy.deepcopy(coord0)
+                while self.test_survived(coord)[0] and coord[p]> -1.0:
+                    coord[p] -= step
+                self.dict_def_range[p][0] = coord[p]-step
+                if self.verbose:
+                    print('Computed range for {pl}: [{mi}, {ma}]'.format(pl=p,
+                                                                         mi=self.dict_def_range[p][0],
+                                                                         ma=self.dict_def_range[p][1]))
+
+        # restore initial parallel computation value
+        self.parallel_computation = init_parcomp
+
+        pass
+
+    def test_points_grid(self, compute_limits=False):
+        """
+        define fixed grid of test points
+        the grid of points is stored in a dictionary of coordinates, one array for each dimension
+
+        :param compute_limits calls compute_range to deterine the maximum ranges before defining the grid
+
+        :return:
+        """
+
+        if compute_limits:
+            self.compute_range()
+
+        # define grid
+        if self.grid_mode == 'radial':
+
+            d_ = {'x': [], 'xp': [], 'y': [], 'yp': [], 'delta': [], 'ct': []}
+
+            # find n_points more than 1
+            two_planes = [p for p in self.planes if self.n_points[p] > 1]
+
+            if len(two_planes) != 2:
+                raise IndexError('There are less or more than 2 planes with more than 1 point to scan')
+
+            ellipse_axes = [self.dict_def_range[two_planes[0]][1],
+                            self.dict_def_range[two_planes[1]][1]]
+
+            if self.dict_def_range[two_planes[1]][0] == 0:
+                theta = np.linspace(0, math.pi, self.n_points[two_planes[0]])
+            else:
+                theta = np.linspace(0, 2*math.pi, self.n_points[two_planes[0]])
+
+            ip = 0
+
+            for ind_p, p in enumerate(self.planes):  # must loop all plane to define all 6D ranges (1 point)
```

Comment:
`ind_p` is not used, though I'm not sure it should not be used instead of `ip`. Please check and then, you could use;
`for p in self.planes:`

Reply:
done